python/lib/modeldata/dmod/modeldata/data/item_domain_detector.py

```diff
@@ -0,0 +1,207 @@
+from dmod.core.meta_data import DataDomain, DataFormat, DiscreteRestriction, StandardDatasetIndex, TimeRange
+from dmod.core.common.reader import RepeatableReader
+from dmod.core.exception import DmodRuntimeError
+from dmod.core.dataset import ItemDataDomainDetector
+from pandas import read_csv as pandas_read_csv
+
+from typing import Optional
+from io import StringIO
+import re
+
+from ..hydrofabric.geopackage_hydrofabric import GeoPackageHydrofabric
+
+# Try to do this if ngen-config package is available
+try:
+    import ngen.config.realization
+    __NGEN_CONFIG_INSTALLED = True
+except ModuleNotFoundError:
+    __NGEN_CONFIG_INSTALLED = False
+
+
+class AorcCsvFileDomainDetector(ItemDataDomainDetector, format_type=DataFormat.AORC_CSV):
+    """
+    Subclass for detecting domains of NextGen regridded per-catchment AORC forcing CSV files.
+
+    Instances must be explicitly or implicitly provided an item name along with the item.  Receiving a :class:`Path`
+    object implicitly satisfies this, with the object's ``name`` property used.  This is required because the applicable
+    catchment for data is not within the data itself; the convention is to include the catchment id as part of the file
+    name.
+    """
+
+    _csv_header: str = "Time,RAINRATE,Q2D,T2D,U2D,V2D,LWDOWN,SWDOWN,PSFC"
+    _datetime_format: str = "%Y-%m-%d %H:%M:%S"
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self._item_name is None:
+            raise DmodRuntimeError(f"{self.__class__.__name__} must be passed an item name on init unless item is file")
+
+        self._num_time_steps = None
+
+    def _get_catchment_id(self) -> str:
+        # TODO: test
+        pattern = re.compile('^.*(cat)[_-](\d+)\D.*$')
```

Comment:
`compile` doesn't necessarily help you here since the pattern is only being used once and destroyed. Might be easier to just roll with `re.match(`'^.*(cat)[_-](\d+)\D.*$', self._item_name)`.

Reply:
Plus, given the simplicity of what this regex is accomplishing, it probably faster to use imperative code here.

Reply:
Yeah, I'll switch things to `re.match`.